R/tokens_xptr.R

```diff
@@ -0,0 +1,209 @@
+#' @rdname tokens_xptr
+#' @export
+is.tokens_xptr <- function(x) {
+    identical(typeof(x), "externalptr") && "tokens_xptr" %in% class(x)
+}
+
+#' Convert a tokens object to a tokens_xptr object
+#'
+#' tokens_xptr is a external pointer object to process a large number of
```

Comment:
This needs more description here, about the implications of converting tokens to an xptr type, and what would need to be done in order to work with such objects, including how the workspace is saved.

We could also list the methods here for which this class is now defined, and explain any differences that the user should know.

The examples should show how to go to and from the xptr space and a few manipulations using standard tokens methods in the middle.

Reply:
I can improve the documentation once we fixed the design. Why don't we remove old tutorials in vignettes, so that we can include more important (technological) things? @stefan-mueller can you decide which articles to keep using Google Analytics and updated them?

Reply:
We only have one vignette (`quickstart.Rmd`), so this would classify as an article unless we included it explicitly as a vignette. But it might be time to create one for performance issues and how to optimise performance with large objects. This could introduce xtokens but also discuss multi-threading, and do some comparisons with other packages.

Reply:
I'd suggest at least updating `NEWS.md` for this PR. We can add a performance vignettes later as we near the 4.0 release.

Reply:
Resolving now but still think we need a performance vignette.

Reply:
If we want to pare the articles in the pkgdown site, we should open a separate issue.

Reply:
I can write an article on xptr only after merging this branch.

Reply:
Happy to get involved in revising the existing vignettes at [quanteda.io](https://quantda.io) and editing the new article on xptr and multithreading. Probably worth discussing details in a separate issue.